### ISSUE TITLE
2.5.2 broken with JRuby 1.7.22 in 2.0 mode

### ISSUE THREAD

**Issue by RobinDaugherty** [rubygems/rubygems#1488]
With JRuby 1.7.22 in 2.0 syntax mode, running the `gem` command or the `bundle` command (probably any other Ruby executable) results in the following error:

```
TypeError: can't convert Regexp into String
                 end_with? at org/jruby/RubyString.java:5281
                     path= at /usr/local/rbenv/versions/jruby-1.7.22/lib/ruby/shared/rubygems/path_support.rb:62
                initialize at /usr/local/rbenv/versions/jruby-1.7.22/lib/ruby/shared/rubygems/path_support.rb:35
                     paths at /usr/local/rbenv/versions/jruby-1.7.22/lib/ruby/shared/rubygems.rb:329
                      path at /usr/local/rbenv/versions/jruby-1.7.22/lib/ruby/shared/rubygems.rb:352
                      dirs at /usr/local/rbenv/versions/jruby-1.7.22/lib/ruby/shared/rubygems/defaults/jruby.rb:84
                 stubs_for at /usr/local/rbenv/versions/jruby-1.7.22/lib/ruby/shared/rubygems/specification.rb:850
                 find_name at /usr/local/rbenv/versions/jruby-1.7.22/lib/ruby/gems/shared/gems/bundler-1.10.6/lib/bundler/rubygems_integration.rb:639
                    report at /usr/local/rbenv/versions/jruby-1.7.22/lib/ruby/gems/shared/gems/bundler-1.10.6/lib/bundler/env.rb:23
                      each at org/jruby/RubyArray.java:1613
                    report at /usr/local/rbenv/versions/jruby-1.7.22/lib/ruby/gems/shared/gems/bundler-1.10.6/lib/bundler/env.rb:22
  request_issue_report_for at /usr/local/rbenv/versions/jruby-1.7.22/lib/ruby/gems/shared/gems/bundler-1.10.6/lib/bundler/friendly_errors.rb:66
      with_friendly_errors at /usr/local/rbenv/versions/jruby-1.7.22/lib/ruby/gems/shared/gems/bundler-1.10.6/lib/bundler/friendly_errors.rb:41
                    (root) at /usr/local/rbenv/versions/jruby-1.7.22/lib/ruby/gems/shared/gems/bundler-1.10.6/bin/bundle:18
                      load at org/jruby/RubyKernel.java:1059
                    (root) at /usr/local/rbenv/versions/jruby-1.7.22/bin/bundle:22
```

This affects version 2.5.2 and I've been able to confirm that 2.5.1 works correctly.

I haven't yet tested with other versions of JRuby.

**Comment by segiddins** [comment]
`Regexp#to_s` should be implemented... ?

**Comment by RobinDaugherty** [comment]
The JRuby code in question is here: https://github.com/jruby/jruby/blob/1.7.22/core/src/main/java/org/jruby/RubyString.java#L5296

**Comment by RobinDaugherty** [comment]
But I agree. Before the exception occurs, I'm able to call `puts` with the Regexp object, and it coerces to string just fine.

**Comment by drbrain** [comment]
JRuby's behavior is correct:

```
$ ruby20 -v -e '"foo".end_with?(/o/)'
ruby 2.0.0p648 (2015-12-16 revision 53162) [x86_64-linux]
-e:1:in `end_with?': no implicit conversion of Regexp into String (TypeError)
	from -e:1:in `<main>'
$ ruby23 -v -e '"foo".end_with?(/o/)'
ruby 2.3.0p0 (2015-12-25 revision 53290) [x86_64-linux]
-e:1:in `end_with?': no implicit conversion of Regexp into String (TypeError)
	from -e:1:in `<main>'
```

**Comment by RobinDaugherty** [comment]
I've tracked down the specific change that broke it: https://github.com/rubygems/rubygems/pull/1157

**Comment by segiddins** [comment]
@RobinDaugherty that particular usage has been fixed on master, I believe.

**Comment by segiddins** [comment]
This is fixed in 2.6.0.
